services-operator/beamsqlstatementsetoperator.py

```diff
@@ -0,0 +1,282 @@
+import kopf
+import time
+import os
+import requests
+from enum import Enum
+import flink_util
+
+namespace = os.getenv("OISP_NAMESPACE") or "oisp"
+FLINK_URL = os.getenv("OISP_FLINK_REST") or f"http://flink-jobmanager-rest.{namespace}:8081"
+FLINK_SQL_GATEWAY = os.getenv("OISP_FLINK_SQL_GATEWAY") or f"http://flink-sql-gateway.{namespace}:9000"
+timer_interval_seconds = os.getenv("TIMER_INTERVAL") or 10
+timer_backoff_seconds = os.getenv("TIMER_BACKOFF_INTERVAL") or 10
+timer_backoff_temporary_failure_seconds = os.getenv("TIMER_BACKOFF_TEMPORARY_FAILURE_INTERVAL") or 30
+
+class States(Enum):
+    INITIALIZED = "INITIALIZED",
+    DEPLOYING = "DEPLOYING",
+    DEPLOYMENT_FAILURE = "DEPLOYMENT_FAILURE",
+    RUNNING = "RUNNING",
+    FAILED = "FAILED",
+    CANCELED = "CANCELED",
+    CANCELING = "CANCELING",
+    UNKNOWN = "UNKNOWN"
+
+
+JOB_ID = "job_id"
+STATE = "state"
+
+@kopf.on.create("oisp.org", "v1alpha1", "beamsqlstatementsets")
+def create(body, spec, patch, logger, **kwargs):
+    name = body["metadata"].get("name")
+    namespace = body["metadata"].get("namespace")
+    kopf.info(body, reason="Creating", message=f"Creating beamsqlstatementsets {name} in namespace {namespace}")
+    logger.info(f"Created beamsqlstatementsets {name} in namespace {namespace}")
+    patch.status[STATE] = States.INITIALIZED.name
+    patch.status[JOB_ID] = None
+    return {"createdOn": str(time.time())}
+
+@kopf.on.delete("oisp.org", "v1alpha1", "beamsqlstatementsets", retries=10)
+def delete(body, spec, patch, logger, **kwargs):
+    """
+    Deleting beamsqlstatementsets
+
+    If state is not CANCELING and not CANCELED and job_id defined, refresh status and trigger cancelation if needed
+    If Canceling, refresh state, when canceled, allow deletion,otherwise wait
+    """
+    name = body["metadata"].get("name")
+    namespace = body["metadata"].get("namespace")
+    state = body['status'].get(STATE)
+    job_id = body['status'].get(JOB_ID)
+    if not state == States.CANCELED.name and not state == States.CANCELING.name and job_id:
+        try:
+            refresh_state(body, patch, logger)
+            if patch.status[STATE]:
+                state = patch.status[STATE]
+            if not state == States.CANCELING.name or not state == States.CANCELED.name:
+                flink_util.cancel_job(logger, job_id)
+        except Exception as err:
```

Comment:
This also catches ugly stuff like `SyntaxError`, better be specific and not `except Exception`

Reply:
But it would be obvious in the logs.